### ISSUE TITLE
Simplify C output in states where most symbols are accepted

### ISSUE THREAD

**Issue by jameysharp** [katef/libfsm#122]
Consider the output of `re -pl c '^([^\\\n]|\\t)'`: In one of the states, it lists cases for all 254 symbols that are not `\n`. I would rather see output structured more like:

```c
case '\\': /* transition to expecting a "t" */
default: /* transition to accept-anything state */
case '\n': /* reject */
```

I think of this as treating the symbols which don't transition as an additional group, and looking at the sizes of all the symbol groups. The largest group should get the `default:` label, even if that isn't the "reject" group.

Perhaps this should be another IR strategy? I guess it's a combination of the "complete" and "error" strategies, and perhaps could subsume both? Or maybe the bug here is just that this case should have used the "error" strategy instead of "complete"? :man_shrugging:

**Comment by katef** [comment]
> it lists cases for all 254 symbols that are not \n
This has to be a bug in the IR i think - one of the existing strategies should cater for exactly that.

If you build with `bmake -r DEBUG=1`:
```
; ~/gh/libfsm/build/bin/re -pl c '^([^\\\n]|\\t)'
re: src/libfsm/print/ir.c:422: make_state: Assertion `fsm_iscomplete(fsm, state)' failed.
zsh: abort      ~/gh/libfsm/build/bin/re -pl c '^([^\\\n]|\\t)'
```
either that state should've been made complete, and wasn't. Or `fsm_iscomplete()` is wrong.

**Comment by katef** [comment]
I think this is another instance of #120.

**Comment by katef** [comment]
Confirmed this is the same issue as #120, addressed in PR #128.

```
; ./build/bin/re -pl c '^([^\\\n]|\\t)'    
#include <assert.h>
#include <stdio.h>

int
fsm_main(int (*fsm_getc)(void *opaque), void *opaque)
{
        int c;

        assert(fsm_getc != NULL);

        enum {
                S0, S1, S2
        } state;

        state = S0;

        while (c = fsm_getc(opaque), c != EOF) {
                switch (state) {
                case S0: /* start */
                        switch ((unsigned char) c) {
                        case '\\': state = S2; break;
                        case '\n': return TOK_UNKNOWN;
                        default: state = S1; break;
                        }
                        break;

                case S1: /* e.g. "a" */
                        break;

                case S2: /* e.g. "\\" */
                        switch ((unsigned char) c) {
                        case 't': state = S1; break;
                        default:  return TOK_UNKNOWN;
                        }
                        break;

                default:
                        ; /* unreached */
                }
        }

        /* end states */
        switch (state) {
        case S1: return 0x1; /* "^([^\\\n]|\\t)" */
        default: return EOF; /* unexpected EOF */
        }
}
```
